lib/utils/json.c

```diff
@@ -559,13 +559,18 @@ static ptrdiff_t get_elem_size(const struct json_obj_descr *descr)
 	}
 	case JSON_TOK_OBJECT_START: {
 		ptrdiff_t total = 0;
+		uint32_t align_shift = 0;
 		size_t i;
 
 		for (i = 0; i < descr->object.sub_descr_len; i++) {
 			total += get_elem_size(&descr->object.sub_descr[i]);
+
+			if (descr->object.sub_descr[i].align_shift > align_shift) {
+				align_shift = descr->object.sub_descr[i].align_shift;
+			}
```

Comment:
Is it sufficient to just walk through this object, or do nested objects need to be traversed as well?

Reply:
Should be sufficient, AFAIK the shifts for structs and whole arrays are calculated correctly and we can't have Z_JSON_ELEMENT_DESCR nested directly. I also did some more testing with nested arrays and structs and everything seems to work fine.